docs/visualizing-your-sql-queries/plotting-with-plotly.md

```diff
@@ -1,14 +1,409 @@
+```
+
+This section was covered in detail in the previous tutorial: [Joining Data in SQL](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#load-the-data). We will be using the same data in this tutorial as well.
+
+```{code-cell} ipython3
+import sys
+import plotly.express as px
+
+sys.path.insert(0, "../../")
+import banking  # noqa: E402
+
+_ = banking.MarketData("https://tinyurl.com/jb-bank-m", "expanded_data")
+_.extract_asc_to_csv()
+```
+
+If you ran the above cell, you should have a folder `expanded_data` in your current directory that contains the `.csv` files we will be using. In this tutorial, we will be focusing on three of these files: `loan.csv`, `account.csv`, `district.csv`.
+
+## Load Engine
+
+We now load in our SQL extension that allows us to execute SQL queries in Jupyter Notebooks.
+
+```{important}
+<b>Note</b> Ensure you restart any previous notebook that has the same database name as the one initialized below.
+```
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%reload_ext sql
+# Initiating a DuckDB database named 'bank_data.duck.db' to run SQL queries
+%sql duckdb:///bank_data.duck.db
+```
+
+## Creating Tables
+
+Let's start off with loading three of the eight `.csv` files from the `expanded_data` folder in the current directory to our newly created DuckDB database. Like in the previous tutorial, we will [create a schema](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#creating-a-schema) `s1` in which we will store the tables. Here we use the `CREATE TABLE` syntax in DuckDB to ingest four of the eight `.csv` files. The `read_csv_auto` is a function that helps SQL understand our local `.csv` file for creation into our database.
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+The code above will create three tables in the database schema: `s1.account`, `s1.district`, `s1.loan`.
+
+## Exploring the data
+
+Let's take a look at each table.
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.account
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.district
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.loan
+```
+
+## Bar Plots
+
+Let's create a bar plot in `plotly` using the `s1.district` table. This plot will visualize the count of each `region` in the data set.
+
+First, let's query the count of each region in SQL.
+
+```{code-cell} ipython3
+%%sql --save region_count
+SELECT region, COUNT(*) as count
+FROM s1.district
+GROUP BY region
+ORDER BY count DESC
+```
+
+Then, let's save the CTE as a Pandas DataFrame.
```

Comment:
Can we utilise [autopandas](https://jupysql.ploomber.io/en/latest/api/configuration.html#autopandas) here?

Reply:
That's a good point. I think autopandas would be useful for students to learn. 

However, a tradeoff is that we lose the use of JupySQL's `--save` feature.  If the purpose of this course is to teach SQL and JupySQL's unique features, then should we keep this workflow?

In my opinion there are several options:
1. Briefly introduce the method we've been using of creating a Pandas DataFrame from CTEs with the `--save` feature **and then** introduce the conveniency of autopandas and move forward using autopandas. (this would require us to change each section's use of CTEs to autopandas)
2. Introduce autopandas in the next section. (given that Pandas will be used)

Thoughts @lfunderburk ?

Reply:
> Briefly introduce the method we've been using of creating a Pandas DataFrame from CTEs with the --save feature and then introduce the conveniency of autopandas and move forward using autopandas. (this would require us to change each section's use of CTEs to autopandas)
> Introduce autopandas in the next section. (given that Pandas will be used)

I think this is a good approach. Instead of waiting for the next section, you could also introduce the two approaches, first `--save` (query 1, plot 1 as an example), then autopandas for the next query.

It would be with the purpose of demonstrating.